src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1330,49 +1330,29 @@ void Lowering::LowerHWIntrinsicCC(GenTreeHWIntrinsic* node, NamedIntrinsic newIn
 void Lowering::LowerFusedMultiplyAdd(GenTreeHWIntrinsic* node)
 {
     assert(node->GetHWIntrinsicId() == NI_FMA_MultiplyAddScalar);
-    GenTreeHWIntrinsic* createScalarOps[3];
+    assert(node->GetOperandCount() == 3);
 
+    bool negatedArgs[3] = {};
     for (size_t i = 1; i <= 3; i++)
     {
         GenTree* arg = node->Op(i);
-
-        if (!arg->OperIsHWIntrinsic() || (arg->AsHWIntrinsic()->GetHWIntrinsicId() != NI_Vector128_CreateScalarUnsafe))
+        if (arg->OperIsHWIntrinsic() && (arg->AsHWIntrinsic()->GetHWIntrinsicId() == NI_Vector128_CreateScalarUnsafe))
```

Comment:
nit, we can make this a bit simpler/more readable:
```suggestion
        if (arg->OperIsHWIntrinsic(NI_Vector128_CreateScalarUnsafe))
```